src/routes/(blog-article)/profiling-the-torrust-bittorrent-udp-tracker/+page.md

```diff
@@ -0,0 +1,328 @@
+
+In the previous article we introduced to the community the benchmarking tools we are using at the moment to know whether the Tracker performs at the same level than other applications on the market.
+
+With those benchmarking tools we ensure that the Tracker has a good performance and we don't have regressions. If you are curious how we do benchmarking you can read the "[Benchmarking the Torrust BitTorrent Tracker](benchmarking-the-torrust-bittorrent-tracker)" article.
+
+In this article we will explain how we are collecting metrics to know which parts of the code we should improve.
+
+## Understating The Tracker
+
+Before continue reading this article you should know a little bit about the tracker internal structure. You can see an overview on the "[Benchmarking the Torrust BitTorrent Tracker](benchmarking-the-torrust-bittorrent-tracker)" article.
+
+<Image src="/images/posts/benchmarking-the-torrust-bittorrent-tracker/torrust-tracker-simplified-architecture.png" alt="Torrust Architecture" />
+
+## Current Challenges
+
+Balancing feature-rich functionality with performance is our central challenge. The **Torrents Repository**, a critical component, has been our focus, serving as the primary bottleneck for scaling request handling capabilities.
+
+Regardless which protocol clients use to connect to the tracker, eventually all requests reach the internal "Tracker" domain service. This tracker contains a repository with the list of all torrents. Each entry on the list contains two pieces of information:
+
+- **The statistics about that torrent**: number of seeders, leechers and peers that have completed downloading.
+- **The peer list**: a list of all the clients announcing the same torrent (swarm).
+
+<CodeBlock lang="json">
+
+```json
+{
+  "info_hash": "090c6d4fb3a03191c4ef1fda6236ef0efb2d5c10",
+  "seeders": 1,
+  "completed": 1,
+  "leechers": 0,
+  "peers": [
+    {
+      "peer_id": {
+        "id": "0x2d71423030303030303030303030303030303031",
+        "client": null
+      },
+      "peer_addr": "0.0.0.0:17548",
+      "updated": 1709916034742,
+      "updated_milliseconds_ago": 1709916034742,
+      "uploaded": 0,
+      "downloaded": 0,
+      "left": 0,
+      "event": "Completed"
+    }
+  ]
+}
+```
+
+</CodeBlock>
+
+The reason why we have been focusing on that part is because we think that's the main bottleneck if we want to increase the number of requests the tracker can handle per second.
+
+The `announce` request is the most important request a tracker needs to handle. Peers get the list of other peers from the tracker by making announce requests. The purpose of that request is:
+
+- To include the peer making the request on the list of peers which are seeding or downloading the torrent.
+- Return the list of other peers so that the client can start asking for torrent pieces to the other peers.
```

Comment:
It would be better to write this sentence this way:
Return the list of other peers so that the client can start asking other peers for torrent pieces.